lib/full-site-editing.php

```diff
@@ -57,6 +57,38 @@ function gutenberg_remove_legacy_pages() {
 
 add_action( 'admin_menu', 'gutenberg_remove_legacy_pages' );
 
+/**
+ * Removes legacy adminbar items from FSE themes.
+ *
+ * @param WP_Admin_Bar $wp_admin_bar The admin-bar instance.
+ */
+function gutenberg_adminbar_items( $wp_admin_bar ) {
+
+	// Early exit if not an FSE theme.
+	if ( ! gutenberg_is_fse_theme() ) {
+		return;
+	}
+
+	// Remove customizer link.
+	$wp_admin_bar->remove_node( 'customize' );
+	$wp_admin_bar->remove_node( 'customize-background' );
+	$wp_admin_bar->remove_node( 'customize-header' );
+	$wp_admin_bar->remove_node( 'widgets' );
+
+	// Add site-editor link.
+	if ( ! is_admin() ) {
```

Comment:
The logic for displaying link is way more complex in WP Core:

https://github.com/WordPress/wordpress-develop/blob/285fa03b3a57629df6a5dc9d92cc7751475c7dee/src/wp-includes/admin-bar.php#L419-L455

We might want to align. If we change it here, we probably should also update how it is exposed in the siebdar menu.

Reply:
The WP-Core implementation is mostly for customizer changesets etc, so not something that can apply here. The only relevant part is the `is_admin` check and the permissions check which is missing here (commit coming in a few minutes)